bundles/automation/org.eclipse.smarthome.automation.module.script/src/main/java/org/eclipse/smarthome/automation/module/script/internal/handler/AbstractScriptModuleHandler.java

```diff
@@ -106,10 +107,13 @@ private void loadConfig() {
     protected void setExecutionContext(ScriptEngine engine, Map<String, ?> context) {
         ScriptContext executionContext = engine.getContext();
 
-        // make the whole context available as "ctx"
+        // Add the rule's UID to the context
+        final Map<String, Object> contextNew = new HashMap<>(context);
+        contextNew.put("ruleUID", this.ruleUID);
+
+        // make the whole (new) context available as "ctx"
         // Note: We don't use "context" here as it doesn't work on all JVM versions!
-        executionContext.setAttribute("ctx", context, ScriptContext.ENGINE_SCOPE);
-        executionContext.setAttribute("ruleUID", this.ruleUID, ScriptContext.ENGINE_SCOPE);
```

Comment:
is it correct that you remove the `ruleUID` from the global context again?

I understand the [comment](https://github.com/eclipse/smarthome/issues/3590#issuecomment-312091953) from @smerschjohann in a way that it should be available in **both** contexts, i.e. the local and the global one.